src/durableorchestrationclient.ts

```diff
@@ -410,29 +428,42 @@ export class DurableOrchestrationClient {
      * @returns A response containing the current state of the entity.
      */
     public async readEntityState<T>(entityId: EntityId, taskHubName?: string, connectionName?: string): Promise<EntityStateResponse<T>> {
-        if (!(this.clientData.baseUrl && this.clientData.requiredQueryStringParameters)) {
-            throw new Error("Cannot use the readEntityState API with this version of the Durable Task Extension.");
-        }
+        let requestUrl: string;
+        if (this.clientData.rpcBaseUrl) {
+            // Fast local RPC path
+            let path = `entities/${entityId.name}/${entityId.key}`;
+            const query: string[] = [];
+            if (taskHubName) { query.push(`taskHub=${taskHubName}`); }
+            if (connectionName) { query.push(`connection=${connectionName}`); }
+
+            if (query.length > 0) {
+                path += "?" + query.join("&");
+            }
 
-        const requestUrl = WebhookUtils.getReadEntityUrl(this.clientData.baseUrl,
-            this.clientData.requiredQueryStringParameters,
-            entityId.name,
-            entityId.key,
-            taskHubName,
-            connectionName);
-
-        try {
-            const response = await this.axiosInstance.get(requestUrl);
-            switch (response.status) {
-                case 200:   // entity exists
-                    return new EntityStateResponse(true, response.data as T);
-                case 404:   // entity does not exist
-                    return new EntityStateResponse(false, undefined);
-                default:
-                    return Promise.reject(new Error(`Webhook returned unrecognized status code ${response.status}`));
+            requestUrl = new URL(path, this.clientData.rpcBaseUrl).href;
+        } else {
+            // Legacy app frontend path
+            if (!(this.clientData.baseUrl && this.clientData.requiredQueryStringParameters)) {
+                throw new Error("Cannot use the readEntityState API with this version of the Durable Task Extension.");
             }
-        } catch (error) {   // error object is axios-specific, not a JavaScript Error; extract relevant bit
-            throw error.message;
+
+            requestUrl = WebhookUtils.getReadEntityUrl(
+                this.clientData.rpcBaseUrl || this.clientData.baseUrl,
```

Comment:
We already know this.clientData.rpcBaseUrl is falsy, so I think we should remove this check.

Reply:
Good catch. I think this is leftover from an earlier attempt to unify the two paths (which I've since abandoned to minimize risk of regression).